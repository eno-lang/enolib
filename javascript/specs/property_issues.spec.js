const enolib = require('..');

describe('Property issues', () => {
  describe('Section', () => {
    describe('toString as field key', () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('toString: ok');

        expect(document.raw()).toMatchSnapshot();
      });
    });

    describe('toString as fieldset key', () => {
      it('does not have any side effects', () => {
        const document = enolib.parse(`
          toString:
          check = ok
        `);

        expect(document.raw()).toMatchSnapshot();
      });
    });

    describe('toString as fieldset entry', () => {
      it('does not have any side effects', () => {
        const document = enolib.parse(`
          check:
          toString = ok
        `);

        expect(document.raw()).toMatchSnapshot();
      });
    });

    describe('fetching from a missing toString field', () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('');

        expect(document.field('toString').optionalStringValue()).toBe(null);
      });
    });

    describe("fetching a missing fieldset with the key 'toString'", () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('');

        expect(document.optionalFieldset('toString')).toBe(null);
      });
    });

    describe("fetching a missing list with the key 'toString'", () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('');

        expect(document.list('toString').optionalStringValues()).toEqual([]);
      });
    });

    describe("fetching a missing section with the key 'toString'", () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('');

        expect(document.optionalSection('toString')).toBe(null);
      });
    });

    describe("fetching missing sections with the key 'toString'", () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('');

        expect(document.sections('toString')).toEqual([]);
      });
    });

    describe('asserting toString has been touched', () => {
      it('does not have any side effects', () => {
        const document = enolib.parse('');

        expect(() => document.assertAllTouched({ only: 'toString' })).not.toThrow();
      });
    });

  });

  describe('Fieldset', () => {

    describe('asserting toString has been touched', () => {
      it('does not have any side effects', () => {
        const document = enolib.parse(`
color ratings:
red = 1
blue = 2
        `.trim());

        const fieldset = document.fieldset('color ratings');

        expect(() => fieldset.assertAllTouched({ only: 'toString' })).not.toThrow();
      });
    });

    describe("fetching a missing fieldset entry with the key 'toString'", () => {
      it('does not have any side effects', () => {
        const document = enolib.parse(`
color ratings:
red = 1
blue = 2
        `.trim());

        const fieldset = document.fieldset('color ratings');

        expect(fieldset.entry('toString').optionalStringValue()).toBe(null);
      });
    });

  });
});
