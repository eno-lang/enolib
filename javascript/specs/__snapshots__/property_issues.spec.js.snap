// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property issues Section toString as field key does not have any side effects 1`] = `
Object {
  "elements": Array [
    Object {
      "key": "toString",
      "type": "field",
      "value": "ok",
    },
  ],
  "type": "document",
}
`;

exports[`Property issues Section toString as fieldset entry does not have any side effects 1`] = `
Object {
  "elements": Array [
    Object {
      "entries": Array [
        Object {
          "key": "toString",
          "type": "fieldsetEntry",
          "value": "ok",
        },
      ],
      "key": "check",
      "type": "fieldset",
    },
  ],
  "type": "document",
}
`;

exports[`Property issues Section toString as fieldset key does not have any side effects 1`] = `
Object {
  "elements": Array [
    Object {
      "entries": Array [
        Object {
          "key": "check",
          "type": "fieldsetEntry",
          "value": "ok",
        },
      ],
      "key": "toString",
      "type": "fieldset",
    },
  ],
  "type": "document",
}
`;
